shelley/chain-and-ledger/formal-spec/Properties.md

```diff
@@ -198,6 +229,11 @@ The size of the mappings PPUpdate, inside the update state, is always at most (n
 **Property**
 The size of the mappings AVUpdate, inside the update state, is always at most num-genesis.
 
+# Epoch Boundary Transition Properties
+
+**Property** The `SNAP`, `POOLREAP`, `NEWPP`, and `EPOCH` transitions 
+are never blocked (i.e. always invoked at the epoch boundary).
```

Comment:
just to make sure that I follow, the justification for this being true is that the `NEWEPOCH` transition calls `EPOCH` on the epoch boundary (ie when`e = e_last + 1`), right? (since `EPOCH` calls the others in every case).

I guess there is actually a (known!) case where this is not true, namely when we don't get a single block in an entire epoch.

besides the situation where the network is down for 5-ish days (in which case doing nothing is probably best), it seems pretty clear that this property is true, but is probably still good to state! thanks!